src/pages/Dashboards/dashboards/shared/Share/Share.js

```diff
@@ -0,0 +1,58 @@
+import React, { useRef, useState } from 'react'
+import cx from 'classnames'
+import { copy } from 'webkit/utils'
+import Svg from 'webkit/ui/Svg/react'
+import ShareModalTrigger from '../../../../../components/Share/ShareModalTrigger'
+import { useShortShareLink } from '../../../../../components/Share/hooks'
+import styles from './Share.module.scss'
+
+const Share = ({ id }) => {
+  const [isShareOpened, setIsShareOpened] = useState(false)
+  const clearTimerRef = useRef()
+  const { shortShareLink, getShortShareLink } = useShortShareLink()
+
+  function onShareClick() {
+    getShortShareLink(window.location.pathname)
+    setIsShareOpened(true)
+  }
+
+  function onCopyLinkClick() {
+    if (clearTimerRef.current) clearTimerRef.current()
+
+    getShortShareLink(window.location.pathname).then((url) => {
+      const node = document.getElementById(id)
+      const clb = () => node && (node.ariaLabel = 'Copy link')
+
+      if (node) node.ariaLabel = 'Copied!'
+      clearTimerRef.current = copy(url, clb)
+    })
+  }
```

Comment:
1. Short link is not needed;
2. `event.currentTarget`
3. Clear timer on unmount